routes/marketdata.js

```diff
@@ -1,8 +1,61 @@
-var express = require('express');
-var router = express.Router();
+const express = require('express');
+const firebase = require('firebase');
+const config = require('./config.js');
+const router = express.Router();
 
-router.get('/', function(req, res, next) {
-	res.render('marketdata');
+if (!firebase.apps.length) {
+    firebase.initializeApp(config.config);
+}
+// Creates connection to database.
+const db = firebase.database();
+// Links to head of database.
+const ref = db.ref("live_weller");
+// Links to markets list.
+const marketsRef = ref.child("markets");
+
+router.get('/:marketName', function(req, res, next) {
+    const marketName = req.params.marketName;
+
+    marketsRef.once('value', function(snapshot) {
+        snapshot.forEach(function(childSnapshot) {
+            const childData = childSnapshot.val();
+            
+            let status = "";
+            let questions = []
+            let missed = [];
+
+            if (childData.marketInfo.marketName === marketName) {
+                status = "Questions to fix to get to level " + (parseInt(childData.marketInfo.marketLevel) + 1);
+                
+                if (childData.marketInfo.marketLevel === 3) {
+                    status = "Market is at Top Level!";
+                }
+
+                for (let key in childData.questions) {
+                    if (key !== "undefined") {
+                        questions.push({key: key, answer: childData.questions[key]});
```

Comment:
This block (and the `childData.missedQuestions` one below) could be a little tighter by using built-in functional programming concepts like [`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) and [`map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map):
```javascript
// so this block...
let questions = [];
for (const key in childData.questions) {
    if (key !=== "undefined") {
        questions.push({ key, answer: childData.questions[key] });
    }
}

// ...can be refactored into this
const questions = childData.questions
    .filter(key => key !== "undefined")
    .map(key => { key, answer: childData.questions[key] });
```

Not 100% sure the above's correct but it's in the ballpark.

Reply:
I'll keep that in mind and give it an edit, thanks!